src/AiClient.php

```diff
@@ -0,0 +1,373 @@
+    {
+        return $availability->isConfigured();
+    }
+
+    /**
+     * Creates a new prompt builder for fluent API usage.
+     *
+     * This method will return an actual PromptBuilder instance once PR #49 is merged.
+     * The traditional API methods in this class will then delegate to PromptBuilder
+     * rather than implementing their own generation logic.
+     *
+     * @since n.e.x.t
+     *
+     * @param Prompt $prompt Optional initial prompt content.
+     * @return PromptBuilder The prompt builder instance.
+     */
+    public static function prompt($prompt = null): PromptBuilder
+    {
+        return new PromptBuilder(self::defaultRegistry(), $prompt);
+    }
+
+    /**
+     * Generates content using a unified API that automatically detects model capabilities.
+     *
+     * When no model is provided, this method delegates to PromptBuilder for intelligent
+     * model discovery based on prompt content and configuration. When a model is provided,
+     * it infers the capability from the model's interfaces and delegates to the capability-based method.
+     *
+     * @since n.e.x.t
+     *
+     * @param Prompt $prompt The prompt content.
+     * @param ModelInterface|null $model Optional specific model to use.
+     * @return GenerativeAiResult The generation result.
+     *
+     * @throws \InvalidArgumentException If the provided model doesn't support any known generation type.
+     * @throws \RuntimeException If no suitable model can be found for the prompt.
+     */
+    public static function generateResult($prompt, ?ModelInterface $model = null): GenerativeAiResult
+    {
+        // If no model provided, use PromptBuilder's intelligent model discovery
+        if ($model === null) {
+            return self::prompt($prompt)->generateResult();
+        }
+
+        // Infer capability from model interface (priority order matters)
+        if ($model instanceof TextGenerationModelInterface) {
+            return self::generateResultWithCapability($prompt, CapabilityEnum::textGeneration(), $model);
+        }
+
+        if ($model instanceof ImageGenerationModelInterface) {
+            return self::generateResultWithCapability($prompt, CapabilityEnum::imageGeneration(), $model);
+        }
+
+        if ($model instanceof TextToSpeechConversionModelInterface) {
+            return self::generateResultWithCapability($prompt, CapabilityEnum::textToSpeechConversion(), $model);
+        }
+
+        if ($model instanceof SpeechGenerationModelInterface) {
+            return self::generateResultWithCapability($prompt, CapabilityEnum::speechGeneration(), $model);
+        }
+
+        throw new \InvalidArgumentException(
+            sprintf(
+                'Model "%s" must implement at least one supported generation interface ' .
+                '(TextGeneration, ImageGeneration, TextToSpeechConversion, SpeechGeneration)',
+                $model->metadata()->getId()
+            )
+        );
```

Comment:
We should always rely on `PromptBuilder` here, even if a `ModelInterface` is provided:
* If the second parameter is a `ModelConfig`, we need to pass it to the constructor of `PromptBuilder`.
* Let's simply take the same one line of code that's inside of the `AiClient::prompt()` method, instead of calling that method. This way we don't need to modify the `AiClient::prompt` signature for something we don't really care about in the fluent API.
* If a `ModelInterface` is provided, we need to call `PromptBuilder::usingModel` before calling `PromptBuilder::generateResult`.

Reply:
I updated this to work like the other methods where it simply relies on the `PromptBuilder` to figure things out. Happily, it did point out a bug in the builder where it didn't infer the capability from the model if provided, which was resolved in 403e3f3.